core/embed/io/nrf/stm32u5/nrf.c

```diff
@@ -509,4 +515,47 @@ bool nrf_system_off(void) {
   return true;
 }
 
+#ifdef SECURE_MODE
+bool nrf_authenticate(void) {
+  nrf_driver_t *drv = &g_nrf_driver;
+  if (!drv->initialized) {
+    return false;
+  }
+
+  drv->info_valid = false;
+
+  uint8_t data[1 + 32] = {MGMT_CMD_CHALLENGE};
+
+  uint32_t challenge[8] = {0};
```

Comment:
Let's not use magic constants.
32 = SHA256_DIGEST_LENGTH
8 = SHA256_DIGEST_LENGTH/sizeof(uint32_t)
or if we switch these two lines then 32 = sizeof(challenge)

Reply:
You're right, the challenge doesn't have to be SHA256_DIGEST_LENGTH :man_facepalming:. That's exactly why magic constants are so deceptive.